apps/cowswap-frontend/src/legacy/state/orders/updaters/UnfillableOrdersUpdater.ts

```diff
@@ -190,17 +190,23 @@ export function UnfillableOrdersUpdater(): null {
     verifiedQuotesEnabled,
   ])
 
+  const updatePendingRef = useRef(updatePending)
   useEffect(() => {
+    updatePendingRef.current = updatePending
+  }, [updatePending])
+
+  useEffect(() => {
+    console.log('[UnfillableOrdersUpdater] Start interval', { updatePendingRef, chainId, account, isWindowVisible })
     if (!chainId || !account || !isWindowVisible) {
       console.debug('[UnfillableOrdersUpdater] No need to fetch unfillable orders')
       return
     }
 
     console.debug('[UnfillableOrdersUpdater] Periodically check for unfillable orders')
-    updatePending()
-    const interval = setInterval(updatePending, PENDING_ORDERS_PRICE_CHECK_POLL_INTERVAL)
+    updatePendingRef.current()
+    const interval = setInterval(updatePendingRef.current, PENDING_ORDERS_PRICE_CHECK_POLL_INTERVAL)
     return () => clearInterval(interval)
-  }, [updatePending, chainId, account, isWindowVisible])
+  }, [updatePendingRef, chainId, account, isWindowVisible])
```

Comment:
In this case, I think the deps for this useEffect should use the same deps as the callback creating the fn, otherwise we run the risk of running an outdated fn, no?

Or potentially move other parts to a ref, such as the balances

Reply:
> In this case, I think the deps for this useEffect should use the same deps as the callback creating the fn, otherwise we run the risk of running an outdated fn, no?

that was what we had before. The deps creating new functions, which eagerly executes the update. This PR just keeps track of the latest function, and when its time to execute, it will invoke it

I don't think its an issue what you mention, and worse case, next check will update the new one

Reply:
I think it'll be because the check depends on the balances.

Here, the interval is created when chainId, account or isWindowVisible changes.
Even if the updatePendingRef.current fn changes, it won't be picked up unless one of those 3 vars change.

We need to know whether the balance changed for the verified quote flag

![image](https://github.com/cowprotocol/cowswap/assets/43217/e95015ea-a119-4f5f-be98-72475657a8e9)

With this change, we can keep querying verified quotes without the user having enough balance.

Reply:
I still don't get your suggestion. 

Maybe we can talk in person since we are sitting close today :P 

We could improve further this to not render the update function so often (by using a ref also for the balances). But the issue this PR tries to solve, is the unnecesary reset of the timers for this polling logic. 

Let's discus it syncronously :)